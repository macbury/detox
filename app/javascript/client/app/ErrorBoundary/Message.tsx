import React, { useEffect } from 'react'
import { Button, Text } from 'react-native'
import styled from 'styled-components/native'

const Container = styled.View`
  flex: 1;
  height: 100%;
  display: flex;
  justify-content: center;
  background: ${(props) => props.theme.colors.background};
`

const Inner = styled.View`
  align-items: center;
`

export interface IMessageProps {
  error: Error
  children?: any
  onRestart()
}

export default function Message({ error, onRestart, children } : IMessageProps) {
  const e = error

  if (e) {
    return (
      <Container>
        <Inner>
          <Text>Oops!</Text>
          <Text>There's an error</Text>
          <Text numberOfLines={10}>{e?.message?.toString()}</Text>
          <Button onPress={onRestart} title="Restart app" />
        </Inner>
      </Container>
    )
  }

  return children
}