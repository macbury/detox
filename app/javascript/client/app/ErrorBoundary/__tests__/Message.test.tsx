import React from 'react'
import { shallow } from 'enzyme'
import { Text } from 'react-native'
import snapshot from 'enzyme-to-json'
import Message, { IMessageProps } from '../Message'

it('renders correctly', () => {
  const props : IMessageProps = {
    error: new Error('Bom bom bommm!'),
    onRestart: jest.fn()
  }
  const component = shallow((
    <Message {...props}>
      <Text>Inner content</Text>
    </Message>
  ))
  expect(snapshot(component)).toMatchSnapshot()
})