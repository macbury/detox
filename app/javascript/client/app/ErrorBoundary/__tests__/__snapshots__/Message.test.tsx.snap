// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Message__Container>
  <Message__Inner>
    <Text>
      Oops!
    </Text>
    <Text>
      There's an error
    </Text>
    <Text
      numberOfLines={10}
    >
      Bom bom bommm!
    </Text>
    <Button
      onPress={[MockFunction]}
      title="Restart app"
    />
  </Message__Inner>
</Message__Container>
`;
